extensions/elytron-security-ldap/deployment/src/main/java/io/quarkus/elytron/security/ldap/deployment/ElytronSecurityLdapProcessor.java

```diff
@@ -61,7 +61,11 @@ ElytronPasswordMarkerBuildItem marker(LdapSecurityRealmBuildTimeConfig ldapSecur
     }
 
     @BuildStep
-    ReflectiveClassBuildItem enableReflection() {
-        return new ReflectiveClassBuildItem(true, true, QuarkusDirContextFactory.INITIAL_CONTEXT_FACTORY);
+    void registerForReflection(BuildProducer<ReflectiveClassBuildItem> reflection) {
+        // All JDK provided InitialContextFactory impls via the module descriptors:
+        // com.sun.jndi.ldap.LdapCtxFactory, com.sun.jndi.dns.DnsContextFactory and com.sun.jndi.rmi.registry.RegistryContextFactory
+        reflection.produce(new ReflectiveClassBuildItem(true, true, QuarkusDirContextFactory.INITIAL_CONTEXT_FACTORY));
```

Comment:
Personal preference: I would prefer that to be

```suggestion
        reflection.produce(new ReflectiveClassBuildItem(true, true, "com.sun.jndi.ldap.LdapCtxFactory"));
```